Assets/Scripts/Mixed/GameObjects/SharedMaterials.cs

```diff
@@ -25,20 +28,31 @@ public void Awake()
             InitializeDictionary();
         }
 
+        public void OnDestroy()
+        {
+            materialDictionary.Clear();
+            materialKeyByNameDictionary.Clear();
+            Instance = null;
+        }
+
         private void InitializeDictionary()
         {
             materialDictionary = new Dictionary<int, Material>();
+            materialKeyByNameDictionary = new Dictionary<string, int>();
+
+            var orderedMaterials = MaterialLibrary.SharedMaterials.OrderBy(m => m.Material.name).ToList();
 
             // Add materials to a dictionary to make querying faster.
-            foreach (var materialLib in MaterialLibrary.SharedMaterials)
-            {
-                materialDictionary.Add(materialLib.MaterialId, materialLib.Material);
+            for(var i = 0; i < orderedMaterials.Count; i++){
+                var material = orderedMaterials[i];
+                materialKeyByNameDictionary.Add(material.Material.name, i);
```

Comment:
Does this support adding more materials dynamically? just a question for future ideas :)

Reply:
I guess it could work, but we would have to rebuild the blob asset reference every time (which could also be done and shouldn't be much trouble)